plugins/Highway_Lanes.py

```diff
@@ -101,12 +105,23 @@ def way(self, data, tags, nds):
                     unknown = False
                     i = 0
                     for tt in ttt:
-                        for t in set(tt.split(";")):
+                        settt = set(tt.split(";"))
+                        for t in settt:
                             if t not in ["left", "slight_left", "sharp_left", "through", "right", "slight_right", "sharp_right", "reverse", "merge_to_left", "merge_to_right", "none", ""]:
                                 unknown = True
                                 err.append({"class": 31606, "subclass": 0 + stablehash64(tl + '|' + t + '|' + str(i)), "text": T_("Unknown turn lanes value \"{0}\"", t)})
-                            if (t == "merge_to_left" and i == 0) or (t == "merge_to_right" and i == len(ttt) - 1):
-                                err.append({"class": 31600, "subclass": 1 + stablehash64(tl + '|' + t + '|' + str(i))})
+
+                        if ("merge_to_left" in settt and i == 0) or ("merge_to_right" in settt and i == len(ttt) - 1):
+                            # A lane must exist in the merging direction
+                            err.append({"class": 31600, "subclass": 1 + stablehash64(tl + '|' + tt + '|' + str(i))})
+
+                        elif (not unknown and len(settt) > 1 and
+                              ((len(settt) > 2 and ("merge_to_right" in settt or "merge_to_left" in settt)) or
+                               ("merge_to_right" in settt and not "merge_to_left" in settt) or
+                               ("merge_to_left" in settt and not "merge_to_right" in settt))):
+                            # a combination of merge_to_* with a turn (other than another merge_to_*)
```

Comment:
Need to check other value are turn. It could be `through`, `none`...

Reply:
I'm sorry, but I don't understand what you mean. It's checked vs `not unknown`, isn't that what you mean?

Reply:
I've renamed the variable `unknown` in the hope that that makes things more clear?

Reply:
`through`, `none` are known value, but there are no turn.

Reply:
But `merge_to_right;through|through` would be an non-existing combination, correct? Just like `merge_to_right;left|right` or anything else. That's just a "through" or a "left" lane, respectively?

And `merge_to_right;none|through` would even be invalid (how can a lane be both "none" and "something" (where something in this example is `merge_to_right`). (Same for `merge_to_right;|through`)

Reply:
I added an explicit check versus a lane containing `none;anything`, where `anything` could be one of `left/slight_left/sharp_left/through/reverse/right/sharp_right/slight_right/merge_to_left/merge_to_right`